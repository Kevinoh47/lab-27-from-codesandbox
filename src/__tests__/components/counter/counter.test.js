import React from 'react';
import renderer from "react-test-renderer";
import Counter from "../../../../src/components/counter/counter.js";   //hard coded to avoid .env issue

describe("counter", () => {
  it("is alive at application start", () => {
    let app = shallow(<Counter />);
    expect(app.find("section").exists()).toBeTruthy();
  });

  it("changes state by decrementing the count when the down link is clicked", () => {
    let app = mount(<Counter />);
    let downClick = app.find("a.downClicker");
    downClick.simulate("click");
    expect(app.state("count")).toEqual(-1);
  });

  it("changes state by incrementing the count when the up link is clicked", () => {
    let app = mount(<Counter />);
    let downClick = app.find("a.upClicker");
    downClick.simulate("click");
    expect(app.state("count")).toEqual(1);
  });

  // this test cannot run on codesandbox.io
  it("renders, and matches snapshot", () => {
    const tree = renderer.create(<Counter />);
    expect(tree).toMatchSnapshot();
  });
});
