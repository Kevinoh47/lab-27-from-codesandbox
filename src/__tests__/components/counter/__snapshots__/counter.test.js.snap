// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter renders, and matches snapshot 1`] = `
<section
  className="counter"
>
  <a
    className="downClicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="upClicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
